test/net/sourceforge/kolmafia/request/RelayRequestTest.java

```diff
@@ -57,4 +72,259 @@ public void exerciseSomeObjectMethodsForCoverage() {
     assertFalse(rr.retryOnTimeout());
     rr.constructURLString("diary.php?textversion=1");
   }
+
+  @Nested
+  class JsonApi {
+    private RelayRequest makeApiRequest(String bodyString) {
+      var rr = new RelayRequest(false);
+      rr.constructURLString("KoLmafia/jsonApi", true);
+      rr.addFormField("pwd", GenericRequest.passwordHash);
+      rr.addFormField("body", bodyString);
+      rr.run();
+      return rr;
+    }
+
+    @BeforeAll
+    public static void beforeAll() {
+      Preferences.reset("RelayRequestTest.ApiRequest");
+      AdventureSpentDatabase.resetTurns();
+      AdventureQueueDatabase.resetQueue();
+    }
+
+    @Test
+    public void returnsProperties() {
+      var cleanups = withProperty("kingLiberated", true);
+      try (cleanups) {
+        var rr =
+            this.makeApiRequest("""
+          { "properties": ["kingLiberated"] }
+          """);
+
+        JSONObject expected =
+            JSON.parseObject("""
+          { "properties": ["true"] }
+          """);
+        assertThat(rr.statusLine, is("HTTP/1.1 200 OK"));
+        assertThat(rr.responseCode, is(200));
+        assertThat(JSON.parse(rr.responseText), is(expected));
+      }
+    }
+
+    @Test
+    public void returnsFunctions() {
+      var cleanups = withTurnsPlayed(22);
```

Comment:
this would correspond to `total_turns_played`, so either don't set this or test that function (preferably the latter).